atomicapp/providers/openshift.py

```diff
@@ -571,3 +622,71 @@ def _get_config_values(self):
         return (result["providerapi"],
                 result["access_token"],
                 result["namespace"])
+
+    def extract(self, image, src, dest, update=True):
+        """
+        Extract contents of a container image from 'src' in container
+        to 'dest' in host.
+
+        Args:
+            image (str): Name of container image
+            src (str): Source path in container
+            dest (str): Destination path in host
+            update (bool): Update existing destination, if True
+        """
+        cleaned_image_name = Utils.sanitizeName(image)
+        pod_name = '{}-{}'.format(cleaned_image_name, Utils.getUniqueUUID())
+        container_name = cleaned_image_name
+
+        # Pull (if needed) image and bring up a container from it
+        # with 'sleep 3600' entrypoint, just to extract content from it
+        artifact = {
+            'apiVersion': 'v1',
+            'kind': 'Pod',
+            'metadata': {
+                'name': pod_name
+            },
+            'spec': {
+                'containers': [
+                    {
+                        'image': image,
+                        'command': [
+                            'sleep',
+                            '3600'
+                        ],
+                        'imagePullPolicy': 'IfNotPresent',
+                        'name': container_name
+                    }
+                ],
+                'restartPolicy': 'Always'
+            }
+        }
+        self.oc.deploy(self._get_url(self.namespace, 'Pod'), artifact)
+        # FIXME: Pod takes some time to come up, so we wait for 3 seconds
+        # The best solution would be to watch for the pod to come to
+        # the running state
+        time.sleep(3)
+
+        # Archive content from the container and return the binary data for
+        # for the archive
+        resp = self.oc.execute(
+            self.namespace, pod_name, container_name,
+            'tar -cz {}'.format('/' + src)
+        )
+        tmpfile = '/tmp/atomicapp-{pod}.tar.gz'.format(pod=pod_name)
+        tmpdir = '/tmp/atomicapp-{pod}'.format(pod=pod_name)
```

Comment:
So, when we do a ``tar -cz /application-entity``, the tar contents start from ``application-entity`` as root dir. And when extracting, it extracts with ``application-entity`` as root directory, even when trying to extract a single member in the tarfile. I could not figure out a simple command to tar the contents of ``/application-entity`` directory, so that we don't need a tmpdir to copy the content from. But I couldn't.

It will be great if you can suggest some work around.